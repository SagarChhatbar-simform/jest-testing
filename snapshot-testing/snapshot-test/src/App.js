import './App.css';
import React from 'react';
import { Component } from 'react';


class App extends Component {
  render() {
    return (
      <div className="App">
        <a href='https://jestjs.io/docs/snapshot-testing'>jest snapshot</a>
      </div>
    );
  }
}


export default App;
