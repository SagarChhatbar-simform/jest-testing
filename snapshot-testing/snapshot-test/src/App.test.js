import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'


describe("snapshot testing",()=>{
  
  test("snapshot for App component",()=>{

    const renderedcomponent = renderer.create(<App/>).toJSON();
    expect(renderedcomponent).toMatchSnapshot()


  })

})