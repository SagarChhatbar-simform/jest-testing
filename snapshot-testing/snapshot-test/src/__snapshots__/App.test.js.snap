// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for App component 1`] = `
<div
  className="App"
>
  <a
    href="https://jestjs.io/docs/snapshot-testing"
  >
    jest snapshot
  </a>
</div>
`;
